cli/cmd/capture/download.go

```diff
@@ -4,82 +4,507 @@
+	}
+	return targetContainer, nil
+}
+
+func getCaptureEnvironment(container *corev1.Container) ContainerEnvVars {
+	captureEnv := ContainerEnvVars{}
+
+	for _, env := range container.Env {
+		switch env.Name {
+		case string(captureConstants.CaptureOutputLocationEnvKeyHostPath):
+			captureEnv.hostPath = env.Value
+		case string(captureConstants.NodeHostNameEnvKey):
+			captureEnv.nodeHostName = env.Value
+		case string(captureConstants.CaptureNameEnvKey):
+			captureEnv.captureName = env.Value
+		case string(captureConstants.CaptureStartTimestampEnvKey):
+			captureEnv.captureStartTime = env.Value
 		}
+	}
+
+	return captureEnv
+}
+
+func getCaptureFileName(env ContainerEnvVars) (string, error) {
+	timestamp, err := file.StringToTimestamp(env.captureStartTime)
+	if err != nil {
+		return "", fmt.Errorf("failed to get the timestamp for the capture file name: %w", err)
+	}
+
+	captureFile := file.CaptureFilename{
+		CaptureName:    env.captureName,
+		NodeHostname:   env.nodeHostName,
+		StartTimestamp: timestamp,
+	}
+
+	fileName := captureFile.String() + ".tar.gz"
+	return fileName, nil
+}
 
-		u, err := url.Parse(blobURL)
+func createDownloadPod(ctx context.Context, kubeClient *kubernetes.Clientset, namespace, nodeName, hostPath, jobName string) (*corev1.Pod, error) {
+	podName := jobName + "-download-" + rand.String(5)
+
+	podSpec := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: namespace,
+		},
+		Spec: corev1.PodSpec{
+			NodeName: nodeName,
+			Containers: []corev1.Container{
+				{
+					Name:    "download",
+					Image:   "busybox",
+					Command: []string{"sleep", "3600"},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "host-mount",
+							MountPath: MountPath, // inside container
+						},
+					},
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyNever,
+			Volumes: []corev1.Volume{
+				{
+					Name: "host-mount",
+					VolumeSource: corev1.VolumeSource{
+						HostPath: &corev1.HostPathVolumeSource{
+							Path: hostPath, // on the node
+						},
+					},
+				},
+			},
+		},
+	}
+
+	fmt.Println("Creating download pod to retrieve the files...")
+	_, err := kubeClient.CoreV1().Pods(namespace).Create(ctx, podSpec, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create download pod: %w", err)
+	}
+
+	for {
+		time.Sleep(1 * time.Second)
+		pod, err := kubeClient.CoreV1().Pods(namespace).Get(ctx, podName, metav1.GetOptions{})
 		if err != nil {
-			return errors.Wrapf(err, "failed to parse SAS URL %s", blobURL)
+			return nil, fmt.Errorf("failed to get download pod: %w", err)
 		}
+		if pod.Status.Phase == corev1.PodRunning {
+			return pod, nil
+		}
+		if pod.Status.Phase == corev1.PodFailed {
+			return nil, ErrDownloadPodFailed
+		}
+	}
+}
+
+func createDownloadExec(kubeClient *kubernetes.Clientset, config *rest.Config, pod *corev1.Pod, srcFilePath string) (remotecommand.Executor, error) {
```

Comment:
is it possible to run this in an e2e step? we create clusters and deploy traffic, would be some extra verification

like a capture test here
https://github.com/microsoft/retina/tree/main/test/e2e/scenarios